src/koopman/algorithms.jl

```diff
@@ -135,6 +135,56 @@ function (x::DMDSVD{T})(X::AbstractArray, Y::AbstractArray, U::AbstractArray, B:
     return x(X, Y-B*U), B
 end
 
+
+"""
+$(TYPEDEF)
+
+Approximates the Koopman operator `K` based on the singular value decomposition
+of `X` and `Y` such that:
+
+```julia
+K₁ = Y*V₁*Σ₁*U₁'
+K₂ = X*V₂*Σ₂*U₂'
+K = (K₁*inv(K₂))^(0.5)
+```
+
+where `Y = U₂*Σ₂*V₂'` and `X = U₁*Σ₁*V₁'` are data matrices. The singular value decomposition is truncated via
+the `truncation` parameter, which can either be an `Int` indiciating an index based truncation or a `Real`
+indiciating a tolerance based truncation. Returns a `Eigen` factorization of the operator.
+
+
+# Fields
+$(FIELDS)
+
+# Signatures
+$(SIGNATURES)
+"""
+mutable struct fbDMD{T} <: AbstractKoopmanAlgorithm where T <: Number
+    """Indiciates the truncation"""
+    truncation::T
+end;
+fbDMD() = fbDMD(0.0)
+function (x::fbDMD{T})(X::AbstractArray, Y::AbstractArray) where T <: Real
+    U₁, S₁, V₁ = truncated_svd(X, x.truncation)
+    xone = one(eltype(X))
+    # Computed the reduced operator
+    Sinv₁ = Diagonal(xone ./ S₁)
+    B₁ = Y*V₁*Sinv₁
+    A₁ = U₁'B₁  # forward DMD
+
+    U₂, S₂, V₂ = truncated_svd(Y, x.truncation)
+    yone = one(eltype(Y))
+    # Computed the reduced operator
+    Sinv₂ = Diagonal(yone ./ S₂)
+    B₂ = X*V₂*Sinv₂
+    A₂ = U₂'B₂  # Backward DMD
+
+    Ã = (A₁/A₂)^(0.5)
```

Comment:
This should be sqrt(A2\A1) . 

Currently we won't be able to run this within the tests, since the operator is a little off. However, I was able to recover the first example from the paper ( purely imaginary eigenvalues, up to a sign. ) by this modification.

Reply:
isnt that like `sqrt(inv(A2)*A1)` ? . I thought we need `sqrt(A1*inv(A2))` as per paper1 and 2. Hence i wrote `sqrt(A1/A2)` earlier

Reply:
hey @AlCap23  Any thoughts later on this ?

Reply:
Sorry, I've been  a busy bee with #330 and some other project work. This PR and your work is still appreciated. The next week I'll not be able to do anything productive in terms of software development or algorithm debugging, but I'll be back ;).

In the mean time: Did you have the implementation running before I started my quest for confusing you ?

Reply:
> Sorry, I've been a busy bee with #330 and some other project work. This PR and your work is still appreciated. The next week I'll not be able to do anything productive in terms of software development or algorithm debugging, but I'll be back ;).
> 
> In the mean time: Did you have the implementation running before I started my quest for confusing you ?


@AlCap23 , My implementation that checks out with tests from [PyDMD fbDMD](https://github.com/mathLab/PyDMD/blob/master/tests/test_fbdmd.py)

```julia
A = (1/√3)*[1 1;-1 2];
y = [[.5; 1.]];
for i in 1:99
           push!(y, A*y[end])
end

X= hcat(y...);

prob = DiscreteDataDrivenProblem(X, t = 1:100);

estimator_exact = solve(prob, DMDSVD() , operator_only = true);   # using DMDSVD
estimator_fbDMD = solve(prob, fbDMD() , operator_only = true);    # using fbDMD

julia> isapprox(Matrix(estimator_exact.K), A, atol = 1e-2)
true

julia> isapprox(Matrix(estimator_fbDMD.K), A, atol = 1e-2)
true

julia> isapprox(eigvals(estimator_exact.K), eigvals(A), atol = 1e-2)
true

julia> isapprox(eigvals(estimator_fbDMD.K), eigvals(A), atol = 1e-2)
true

julia> estimator_exact.C ≈ diagm(ones(2))
true

julia> estimator_fbDMD.C ≈ diagm(ones(2))
true

```

I don't understand though why tests linear and nonlinear tests in DataDrivenDiffEq/koopman is not working !

Reply:
for now i have added test for [fbDMD from PyDMD](https://github.com/mathLab/PyDMD/blob/master/tests/test_fbdmd.py)